sdk/compatability/src/lib.rs

```diff
@@ -0,0 +1,694 @@
+    *account_rotated.sequence_number_mut() += 1;
+
+    // Verify that the account has the new authentication key
+    let account_view = client.get_account(account.address())?.into_inner().unwrap();
+    assert_eq!(
+        account_view.authentication_key.inner(),
+        account_rotated.authentication_key().as_ref(),
+    );
+
+    // Verify that we can rotate it back
+    let txn = account_rotated.sign_transaction(
+        env.transaction_factory()
+            .rotate_authentication_key(account.authentication_key())
+            .sender(account_rotated.address())
+            .sequence_number(account_rotated.sequence_number())
+            .build(),
+    );
+
+    client.submit(&txn)?;
+    let txn_view = client
+        .wait_for_signed_transaction(&txn, None, None)?
+        .into_inner();
+
+    assert!(txn_view.vm_status.is_executed(), "{:?}", txn_view.vm_status);
+
+    // Verify that the account has the old auth key again
+    let account_view = client.get_account(account.address())?.into_inner().unwrap();
+    assert_eq!(
+        account_view.authentication_key.inner(),
+        account.authentication_key().as_ref(),
+    );
+
+    Ok(())
+}
+
+#[test]
+#[ignore]
+fn dual_attestation() -> Result<()> {
+    use diem_sdk::crypto::ed25519::ed25519_dalek;
+
+    let env = Environment::from_env();
+    let client = env.client();
+
+    let dual_attestation_limit = client
+        .get_metadata()?
+        .into_inner()
+        .dual_attestation_limit
+        .unwrap();
+    let start_amount = dual_attestation_limit + 1000;
+    let transfer_amount = dual_attestation_limit + 100;
+    let currency = Currency::XUS;
+
+    let mut account = LocalAccount::random();
+    env.coffer()
+        .fund(currency, account.authentication_key(), start_amount)?;
+
+    let account_2 = LocalAccount::random();
+    env.coffer()
+        .fund(currency, account_2.authentication_key(), start_amount)?;
+
+    // Sending a txn that's over the dual_attestation_limit without a signature from the reciever
+    let txn = account.sign_transaction(
+        env.transaction_factory()
+            .peer_to_peer(currency.type_tag(), account_2.address(), transfer_amount)
+            .sender(account.address())
+            .sequence_number(account.sequence_number())
+            .build(),
+    );
+
+    client.submit(&txn)?;
+    let txn_view = client
+        .wait_for_signed_transaction(&txn, None, None)?
+        .into_inner();
+    *account.sequence_number_mut() += 1;
+
+    assert!(
+        !txn_view.vm_status.is_executed(),
+        "{:?}",
+        txn_view.vm_status
+    );
+
+    // Add a dual_attestation creds for account_2
+    let dual_attestation_secret_key = ed25519_dalek::SecretKey::generate(&mut rand::rngs::OsRng);
+    let dual_attestation_extended_secret_key =
+        ed25519_dalek::ExpandedSecretKey::from(&dual_attestation_secret_key);
+    let dual_attestation_public_key = ed25519_dalek::PublicKey::from(&dual_attestation_secret_key);
```

Comment:
nvm, I see where `ed25519_dalek` is needed... also, wow, what is going on here? : 0

random: looked into what an `ExpandedSecretKey` is... this comment is pretty funny (https://github.com/dalek-cryptography/ed25519-dalek/blob/925eb9ea56192053c9eb93b9d30d1b9419eee128/src/secret.rs#L200)

poor isis... https://github.com/dalek-cryptography/ed25519-dalek/blob/925eb9ea56192053c9eb93b9d30d1b9419eee128/src/secret.rs#L462